components/predict-form/Predict.tsx

```diff
@@ -482,20 +381,16 @@ export function Predict({
                   }
                   maxRows={15}
                   onChange={(e) => {
-                    smartUpdateResolveBy(e.currentTarget.value)
                     updateTagsPreview(e.currentTarget.value)
                     void onChangeQuestion(e)
                   }}
-                  onKeyDown={(e) => {
-                    if (onEnterSubmit(e)) return
-                    // current value doesn't include the key just pressed! So
-                    if (e.key.length === 1) {
-                      smartUpdateResolveBy(e.currentTarget.value + e.key)
-                    }
-                  }}
+                  // TODO flag in PR that there might be some case where onKeyDown was doing something different to onChange
+                  onKeyDown={onEnterSubmit}
```

Comment:
It appeared that this block:
```typescript
if (e.key.length === 1) {
    smartUpdateResolveBy(e.currentTarget.value + e.key)
}
```
isn't needed after moving `smartUpdateResolveBy` into a useEffect. Although it's possible I'm missing something (such as it being needed on mobile). It was added quite recently by @adam-binks

Reply:
Nice, that's a much better solution! 

(I found that without this line if you typed "Mon", it would only set it to Monday after you typed another char (e.g. "Monx"). In other words, it used the previous value not the current. Looks like that issue doesn't happen if we useEffect instead, so all good)